test/php/library/Icinga/Crypt/RSATest.php

```diff
@@ -0,0 +1,91 @@
+<?php
+// Icinga Web 2 | (c) 2020 Icinga Development Team | GPLv2+
+
+namespace Tests\Icinga\Crypt;
+
+use Icinga\Crypt\RSA;
+use Icinga\Test\BaseTestCase;
+use InvalidArgumentException;
+use UnexpectedValueException;
+
+class RSATest extends BaseTestCase
+{
+    /**
+     * @expectedException InvalidArgumentException
+     */
+    function testLoadKeyThrowsExceptionIfMoreThanTwoKeysGiven()
+    {
+        (new RSA())->loadKey('one','two','three');
+    }
+
+    /**
+     * @expectedException UnexpectedValueException
+     */
+    function testGetPublicKeyThrowsExceptionIfNoPublicKeySet()
+    {
+        (new RSA())->getPublicKey();
+    }
+
+    /**
+     * @expectedException UnexpectedValueException
+     */
+    function testGetPrivateKeyThrowsExceptionIfNoPrivateKeySet()
+    {
+        (new RSA())->getPrivateKey();
+    }
+
+    function testLoadKeyAutomaticallyDetectsThePublicAndPrivateKey()
+    {
+        list($privateKey, $publicKey) = RSA::keygen();
+
+        $rsa = (new RSA())->loadKey($publicKey, $privateKey);
+        $this->assertSame($privateKey, $rsa->getPrivateKey());
+        $this->assertSame($publicKey, $rsa->getPublicKey());
+
+        $rsa = (new RSA())->loadKey($privateKey, $publicKey);
+        $this->assertSame($privateKey, $rsa->getPrivateKey());
+        $this->assertSame($publicKey, $rsa->getPublicKey());
+    }
+
+    function testEncryptReturnEmptyArrayIfNoParameterGivenAndReturnEncryptedValueIfParameterIsGiven()
```

Comment:
Your tests are a bit odd because of all the `empty` and `array_filter` calls and bool comparisons with `assertSame`.
Testing for bools is done via `assertTrue` and `assertFalse`. If you want to test for the empty array use:

```php
$this->assertSame([], $actual);
```

Tests should try cover exactly one use case, so you should have 4 tests to begin with that only test if the empty array is returned when no arguments are given to the encryption and decryption functions, i.e.:

```php
public function testEncryptReturnsEmptyArrayIfNoArgumentsGiven()
{
    $this->assertSame([], (new RSA())->encrypt();
}
```

The same goes for the other functions/tests.

Now I think it does not make sense to test encryption and decryption isolated but have one test case for that, i.e.:

```php
public function testEncryptionAndDecryption()
{
    $rsa = (new RSA())->loadKey(...RSA::keygen());

    $data = ['foo', 'bar'];

    $this->assertSame($data, $rsa->decrypt(...$rsa->encrypt(...$data)));
}

public function testEncryptionToBase64AndDecryptionFromBase64()
{
    ....
}
```

I just saw that all functions are missing the `public` keyword. I started with the tests so it is kind of my fault 😆 
But please add it.